core/core-workflow-steps/src/main/java/org/eclipse/sw360/antenna/csvreader/CSVReader.java

```diff
@@ -394,11 +399,17 @@ private static String mapCPEIdToString(Artifact artifact) {
         .orElse("");
     }
 
-    private static String mapFileNameToString(Artifact artifact) {
-        return artifact.askFor(ArtifactSourceFile.class)
+    private static String getFilepathAsString(Artifact artifact) {
+        Optional<Path> path = artifact.askFor(ArtifactSourceFile.class)
                 .map(ArtifactFactWithPayload::get)
                 .map(Path::toAbsolutePath)
-                .map(Path::toString)
-                .orElse("");
+                .filter(pth -> pth.toFile().exists());
+        if (path.isPresent()) {
+            return path.get().toString();
+        } else {
+            artifact.getMainCoordinate().ifPresent(coordinate ->
+                    LOGGER.debug("The given source file {} for artifact {} does not exist.", path, coordinate));
```

Comment:
Äh, path.isPresent() is false, what does this log message give, it uses an empty optional.toString, and what in case of a non existing main coordinate, there is still an artifact, how to deal with that?